pandas/tests/frame/test_alter_axes.py

```diff
@@ -186,18 +186,19 @@ def test_set_index_pass_arrays_duplicate(self, frame_of_index_cols, drop,
 
         # == gives ambiguous Boolean for Series
         if drop and keys[0] is 'A' and keys[1] is 'A':
-            with tm.assert_raises_regex(KeyError, '.*'):
-                df.set_index(keys, drop=drop, append=append)
+            # can't drop same column twice
+            first_drop = False
```

Comment:
why are you changing these?

these tests are not very hard to interpret

Reply:
This changes because now we're not failing for `drop=True` and `keys=['A', 'A']` (for example), see #22484

Reply:
this tests is very confusing. pls make it simpler.

Reply:
The diff might be confusing, but it doesn't look so bad in practice: https://github.com/pandas-dev/pandas/blob/e861dcdafb0b2674f6be311b611cf5988c17b1e5/pandas/tests/frame/test_alter_axes.py#L187

It doesn't get much simpler - to test against previously tested behaviour, I have to do the `set_index`-calls sequentially, but it's not possible to drop the same key twice (which this test is changing).